### ISSUE TITLE
Lagging in iOS 16.6 iPhone high frame rates  devices

### ISSUE THREAD

**Issue by xiaoxiaowesley** [flutter/flutter#131319]
8/20/23 - See https://github.com/flutter/flutter/issues/131319#issuecomment-1683070516 below for guidance - @zanderso 

---

On July 25, Apple pushed the iOS 16.6 version. We have received reports from our users that Flutter is stuttering on iPhones with high frame rates under this system. I raised this issue in the hope that more people will pay attention to this issue and work together to solve it.

### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Just run the flutter compare with the lower iOS version.

### Expected results

No stuttering under iOS 16.6

### Actual results

stuttering under iOS 16.6

### Code sample

<details><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

It is clear to see that the iPhone 14, which has better performance, stutters more than the iPhone 12 under the same code.

iPhone 14  iOS 16.6

https://github.com/flutter/flutter/assets/44207000/576815f3-e462-400b-ae36-d80b7a86f62b

Devtool performance trace screenshot

![image](https://github.com/flutter/flutter/assets/44207000/df6ec70c-a453-414a-ba8d-37e05d156590)

iPhone 12 (no high rate) iOS 15.4.1

https://github.com/flutter/flutter/assets/44207000/cbea2fc7-e1ef-4ad4-aa2d-2b4db02ffe50

Devtool performance trace screenshot

![image](https://github.com/flutter/flutter/assets/44207000/049ca1fb-cb5b-4de5-a508-9fdebd228ef2)

</details>

### Logs

<details><summary>Logs</summary>

```console
NO LOG
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[!] Flutter (Channel unknown, 3.3.9-0.0.pre.172, on macOS 13.4.1 22F82 darwin-x64, locale zh-Hans-CN)
    ! Flutter version 3.3.9-0.0.pre.172 on channel unknown at /Users/xiaoxiang/Documents/software/multi_flutter
    ! Upstream repository unknown
[!] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
[✓] Chrome - develop for the web
[!] Android Studio (version 2022.1)
    ✗ Unable to find bundled Java version.
[!] Android Studio (version 2022.1)
    ✗ Unable to find bundled Java version.
[✓] IntelliJ IDEA Ultimate Edition (version 2023.1.2)
[✓] IntelliJ IDEA Ultimate Edition (version 2022.3.2)
[✓] IntelliJ IDEA Ultimate Edition (version 2023.1.2)
[✓] VS Code (version 1.80.1)
```

</details>

**Comment by navaronbracke** [comment]
I see you are running an older version of Flutter. Does this reproduce with the latest master and/or the current stable (version 3.10.6) ?

**Comment by SilverFruity** [comment]
I use flutter 3.10.6 to test in flutter gallery, also has the same problem under iOS 16.6
![DC4863DA-3842-429C-8511-943F32DB252B](https://github.com/flutter/flutter/assets/16136774/d9d220ce-47e6-4a86-be24-e2114c80d6d2)

**Comment by roybeitner** [comment]
I have the same issue.
With and without CADisableMinimumFrameDurationOnPhone true. @navaronbracke 

Before iOS 16.6 it was super smooth, suddenly something flutter in that iOS version.

**Comment by justprodev** [comment]
on iPhone 14 Pro it looks really bad compared to native apps...

like interlacing on an old TV

enabling Settings->Accessibility->Motion->Limit Frame Rate helps

**Comment by naivetoby** [comment]
@navaronbracke 

I have the same problem.

`CADisableMinimumFrameDurationOnPhone` to `true` in my Info.plist

Before iOS 16.6 it was super smooth.

**Comment by roybeitner** [comment]
@dam-ease This issue has to be prioritized, it's super noticeable and a large amount of users are complaining...
The regressions and instability of flutter after iOS updates is getting ridiculous :(

**Comment by luckysmg** [comment]
IOS 16.6 and iOS 17 beta1 beta2 are janky. But iOS17 beta3 is ok... Seems it is a iOS bug...Hope Apple can post a hotfix version for 16.6......

**Comment by m-j-g** [comment]
The complaints are starting to roll in. All other apps run smooth on 16.6 so I’m not sure this is something they would fix. Seems to only be flutter apps.

**Comment by Joan1211** [comment]
I also have the same issue with my flutter app on my iphone 14pro. 120hz looks jaggy.

**Comment by UCetinkaya94** [comment]
This is really bad. This issue needs way more attention.

**Comment by luckysmg** [comment]
I reproduced this issue with a simple ListView.
I found something maybe helpful for this issue.

Below conclusions is tested on iPhone13Pro iOS **16.6** with a simple ListView

❌: lagging
✅：smooth

### 3.10.6 
skia ❌ 
impeller ❌

### 3.13.0 0.2pre 
skia ❌ 
impeller ✅

### master 
skia ❌ 
impeller ✅

I m not sure what impeller done during 3.10.6 to 3.13.0 to make this thing better.

cc @chinmaygarde @jonahwilliams @bdero who maybe know more about with impeller recently changes. ^_^

**Comment by jonahwilliams** [comment]
in 3.13 in Impeller we unconditionally use a CAtransaction to manage presentation

**Comment by luckysmg** [comment]
In impeller use `presentationWithTransation = YES` to avoid junky. 
But skia using `presentsWithTransaction = [[NSThread currentThread] isMainThread];`

**Because in some cases if we present in raster thread will cause crash**. In below image I set `presentsWithTransaction=YES` in skia side and cause this crash in our app.

<img src=https://github.com/flutter/flutter/assets/49340347/bbd0817c-1de2-4560-9259-f724e05cb11c width=700 />

**Comment by jonahwilliams** [comment]
I don't think you're going to be able to patch that change into old or non-impeller based engines.

**Comment by luckysmg** [comment]
Yes.can't patch that change.

**Comment by devberkay** [comment]
Same here. There has been a significant downfall in raster thread performance. The frame rate has jumped up to 15-16 ms from the previous 6-7 ms. It used to be buttery smooth with a steady 120 fps and under 8 ms refresh easily but after the iOS update, it has become very junky.

**Comment by darkalfx** [comment]
This isnt just a bit of stuttering here and there, it effectively kills Flutter on iOS thats how bad this is.

**Comment by huvi8161** [comment]
@chinmaygarde , @jonahwilliams  do you plane to release a fix for 3.10.x(Impeller) soon, also if and when you find a fix for older versions like 3.7.x(legacy renderer) are you going to issue a fix for that also?

**Comment by jonahwilliams** [comment]
The Solution that Impeller is currently using on 3.13 branches is not adaptable to Skia versions of the renderer. We'll discuss a hotfix for the 3.10 Impeller renderer this week but that would likely depend on finishing the release process for the 3.13 stable. If you're able to update to 3.13 beta this is likely the quickest path to a fix.

We have no plans to investigate independent solutions for the Skia renderer on either current or older versions of the framework.

**Comment by huvi8161** [comment]
We have a running production on version 3.7.x(and experiencing this problem with end users) and a testing version on 3.10.x but going to 3.13 beta is not an option for a production application it would be to risky, even 3.10.x is risky, but on test app looks stable and it would be great if you could roll out a fix for 3.10.x this week.

**Comment by chinmaygarde** [comment]
Triage: This is fixed on main and the 3.13 candidates. The 3.13 stable release should be this week and apps needing the fix urgently can use this immediately.

**Comment by notandyvee** [comment]
@chinmaygarde could you elaborate which are the candidates? Is it the beta branch? Because the last change was the 8th. Could be missing something. Knowing exact versions would be helpful.

**Comment by SethuSenthil** [comment]
> @chinmaygarde could you elaborate which are the candidates? Is it the beta branch? Because the last change was the 8th. Could be missing something. Knowing exact versions would be helpful.

I think he means the master branch. I would not recommend cause it’s literally the most cutting edge branch with minimal testing. Definitely stay away for production apps.

**Comment by jonahwilliams** [comment]
We mean beta channel.

**Comment by SethuSenthil** [comment]
Thanks for the clarification!

**Comment by huvi8161** [comment]
> Triage: This is fixed on main and the 3.13 candidates. The 3.13 stable release should be this week and apps needing the fix urgently can use this immediately.

I would again please ask you to reconsider and release a fix for current version 3.10.x, we have a production app, for which users are paying access and we cant’t switch to 13.3.x just like that because our app uses a processing tool(for security) and this tool is also a product that has an a release cycle that depends on Fluttter releases(they can’t issue a support until new Flutter release is out). As I said we have a working support for Flutter 3.10.x and we need a fix for this version and not a new release of Flutter. If you need any other details you can contact me directly on mail I would be happy to explain everything. Thank you!

**Comment by flirtyapp** [comment]
> > Triage: This is fixed on main and the 3.13 candidates. The 3.13 stable release should be this week and apps needing the fix urgently can use this immediately.
> 
> @chinmaygarde , @jonahwilliams I would again please ask you to reconsider and release a fix for current version 3.10.x, we have a production app, for which users are paying access and we cant’t switch to 13.3.x just like that because our app uses a processing tool(for security) and this tool is also a product that has an a release cycle that depends on Fluttter releases(they can’t issue a support until new Flutter release is out). As I said we have a working support for Flutter 3.10.x and we need a fix for this version and not a new release of Flutter. If you need any other details you can contact me directly on mail I would be happy to explain everything. Thank you!

My understanding from the above comment is if you need the update immediately you can use the 3.13 candidates (beta) or wait for the stable (3.10.7) to be released this week. I believe the stable will be 3.10.7.

**Comment by riveraj33** [comment]
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> > > Triage: This is fixed on main and the 3.13 candidates. The 3.13 stable release should be this week and apps needing the fix urgently can use this immediately.
> 
> > 
> 
> > @chinmaygarde , @jonahwilliams I would again please ask you to reconsider and release a fix for current version 3.10.x, we have a production app, for which users are paying access and we cant’t switch to 13.3.x just like that because our app uses a processing tool(for security) and this tool is also a product that has an a release cycle that depends on Fluttter releases(they can’t issue a support until new Flutter release is out). As I said we have a working support for Flutter 3.10.x and we need a fix for this version and not a new release of Flutter. If you need any other details you can contact me directly on mail I would be happy to explain everything. Thank you!
> 
> 
> 
> My understanding from the above comment is if you need the update immediately you can use the 3.13 candidates (beta) or wait for the stable (3.10.7) to be released this week. I believe the stable will be 3.10.7. 

I don't think they mean a 3.10 hot fix will come out. They mention Main and Beta candidate. By main I believe it's referencing the master channel. So this fix is on master and the new stable release based on the 3.13 beta.

**Comment by flirtyapp** [comment]
> > > > Triage: This is fixed on main and the 3.13 candidates. The 3.13 stable release should be this week and apps needing the fix urgently can use this immediately.
> > 
> > 
> > > 
> > 
> > 
> > > @chinmaygarde , @jonahwilliams I would again please ask you to reconsider and release a fix for current version 3.10.x, we have a production app, for which users are paying access and we cant’t switch to 13.3.x just like that because our app uses a processing tool(for security) and this tool is also a product that has an a release cycle that depends on Fluttter releases(they can’t issue a support until new Flutter release is out). As I said we have a working support for Flutter 3.10.x and we need a fix for this version and not a new release of Flutter. If you need any other details you can contact me directly on mail I would be happy to explain everything. Thank you!
> > 
> > 
> > My understanding from the above comment is if you need the update immediately you can use the 3.13 candidates (beta) or wait for the stable (3.10.7) to be released this week. I believe the stable will be 3.10.7.
> 
> I don't think they mean a 3.10 hot fix will come out. They mention Main and Beta candidate. By main I believe it's referencing the master channel. So this fix is on master and the new stable release based on the 3.13 beta.

Ah right when they mentioned stable I just assumed the stable channel and the next update being 3.10.7. Thanks for explaining.

**Comment by notandyvee** [comment]
@chinmaygarde @jonahwilliams apologies for pinging you both. But I tried to move forward with the beta channel (3.13 pre-4). But something is causing my app to hang. No errors. Just hangs. Since there's a chance it's a library that isn't ready to be used with the changes made in 3.13, I'd like to request this fix be added to 3.10 to cast the widest net possible. Unless you tell us the effort is huge to cherry pick those changes, there should be no reason to cause people delays in putting out a quick fix.

**Comment by dnfield** [comment]
> @chinmaygarde @jonahwilliams apologies for pinging you both. But I tried to move forward with the beta channel (3.13 pre-4). But something is causing my app to hang. No errors. Just hangs. Since there's a chance it's a library that isn't ready to be used with the changes made in 3.13, I'd like to request this fix be added to 3.10 to cast the widest net possible. Unless you tell us the effort is huge to cherry pick those changes, there should be no reason to cause people delays in putting out a quick fix.

It would be helpful if you could open a bug about the hang on 3.13. As mentioned, 3.13 is very soon going to be the stable release of Flutter.

**Comment by notandyvee** [comment]
> It would be helpful if you could open a bug about the hang on 3.13. As mentioned, 3.13 is very soon going to be the stable release of Flutter.

I didn't because theres literally no error. Hangs. But I submitted the issue and pray you guys are willing to help me.

**Comment by stuartmorgan** [comment]
Has anyone affected by this (for instance @xiaoxiaowesley or @luckysmg?) filed an issue with Apple via their Feedback Assistant? If so, it would be helpful if you could let me know the feedback ID for reference.

**Comment by SethuSenthil** [comment]
Updating to Flutter 3.13.0 stable seems to have fixed this issue for me!

**Comment by arnab** [comment]
Tried to upgrade to `3.13.0-stable` (along with the latest Dart `3.1.0`) and ran into this issue with a dependency: https://github.com/optimist-dev/scrolls_to_top/issues/11.

**Comment by zanderso** [comment]
Hi all, we were really surprised by this issue coming up and affecting already deployed apps on an iOS minor version release. It's unfortunate that a minor version release from Apple would break deployed apps like this. Normally, it's our policy not to backport fixes to an old stable release, but because of these special circumstances, we are planning to backport the fix for this issue to 3.10 on the stable channel.

Just to reiterate the above comments, the preferred solution is to upgrade to the 3.13 stable release, which does not have this issue. But for those who aren't ready yet, we expect another 3.10 hotfix to be released within the next few days. The hotfix is not without risk. The exact mechanism of the regression remains unknown, and is likely dependent on undocumented behavior of iOS APIs. Additionally, since 3.13 has already been released, obtaining the next 3.10 hotfix will require following a slightly different upgrade path than usual. We'll have instructions available at the time of the hotfix release.

I'd like to give a huge thanks to the release team for bending over backwards to make this happen.

Thanks for your patience, and apologies for any inconvenience!

**Comment by arnab** [comment]
Curious if anyone else is seeing any problems with this release? Our app crashes while playing background audio when it's not in the foreground.

### Details

We have an app that uses:

```
  just_audio: ^0.9.34
  just_audio_background: ^0.0.1-beta.10
```

_(and so also indirectly depends on their dependent audio related packages)._

We upgraded to Flutter `3.13.0-stable` on Wed Aug 16. And since then, **we started seeing crashes (full app crash)** on **iPhone 14 Pro (running iOS 16.6)**. More details on the patterns we have seen below, but essentially:

* we have only seen it fail on iPhone 14 Pros. We have tried two iPhone 14 Pros and it crashes consistently on both!
* we also are testing on iPhone 12 Pro and iPhone 13 Pro, as well as Andriod (Samsung Galaxy A52) - no crashes so far.
* the app seems to crash when running in background mode while audio playing
* `Crashlytics`, or `background_audio`'s error handler are not seeing any of these crashes. We have this code to detect errors from `background_audio` (as well as comprehensive Crashlytics):

```dart
    _player.playbackEventStream.listen((event) {},
        onError: (Object e, StackTrace stackTrace) => errorReportingService
            .reportError('Error in playback stream', e, stackTrace));
```

(_The `errorReportingService` is configured to log it and also send to Crashlytics, and we know it works because we see other errors in Crashlytics, as expected_)

I downgraded Flutter back to `3.10.6-stable` today and have not seen any crashes so far.

To reiterate:

* we did not have crashes (on any device, including iPhone 14 Pros) before AUg 16th (on Flutter `3.10.6-stable`)
* we have consistent crashes on iPhone 14 Pros with the `3.13.0-stable` release
* after downgrading Flutter back to `3.10.6-stable` earlier today (a few hours ago), we have not seen crashes again. (But yes, that terrible lagging/jank is back :()

### Patterns on failures on iPhone 14 Pro + iOS 16.6

From @ilyabezdelev (who is seeing this in two iPhone 14 Pros), patterns when it crashes:

* iPhone locked, audio is playing in the background => the app crashes after a few minutes of playing. Sometimes it crashes a few seconds after locking the phone. Sometimes it may take 20-ish minutes before it crashes.
* iPhone unlocked, switching to another app => the app crashes within a minute or so.
* iPhone unlocked, app is open, auto-lock is set to 5 mins => app crashes immediately after the phone is auto-locked

It does not seem to crash when the app is in foreground.

### Crash reports

Surprisingly, Crashlytics does not show any of these crashes. I have even instrumented everything instructed in Crashlytics docs:

* [uncaught](https://firebase.google.com/docs/crashlytics/customize-crash-reports?hl=en&authuser=0&platform=flutter#report-uncaught-exceptions), including async ones; and [caught exceptions](https://firebase.google.com/docs/crashlytics/customize-crash-reports?hl=en&authuser=0&platform=flutter#report-caught-exceptions)
* and even [Errors outside of Flutter Isolate](https://firebase.google.com/docs/crashlytics/customize-crash-reports?hl=en&authuser=0&platform=flutter#errors-outside-flutter)

We do see TestFlight crash reports, but they don't have any stack-trace or any error message.

### Repro

_PS: Our app is in TestFlight (and not launched yet), so I am unable to share code or more details. However, I will try and also create a minimal repro app if folks want it._

**Comment by notandyvee** [comment]
> Curious if anyone else is seeing any problems with this release? Our app crashes while playing background audio when it's not in the foreground.

Curious indeed @arnab. I am having issues as well. See [issue 132666](https://github.com/flutter/flutter/issues/132666) Anytime I foreground the app for some reason a custom scroll view with multiple slivers causes issues. Obviously different than your situation. But very curious that backgrounding the app causes a crash with no error. When I pushed to testflight, backrounding the app is when it crashes. Constant crash dialogs. The app is essentially unusable. Version 3.10 works fine for me during my tests.

**Comment by arnab** [comment]
Huh - that's a great find @notandyvee !

My app also uses CustomScrollView and Sliver widgets (grids, lists and more). 

So it might have more to do with that, rather than background_audio.

**Comment by jonahwilliams** [comment]
Scrolling/Slivers are part of the framework and not a part of the engine. If you have any native stack traces we'll be happy to take a look of course

**Comment by naivetoby** [comment]
> Curious if anyone else is seeing any problems with this release? Our app crashes while playing background audio when it's not in the foreground.
> 
> ### Details
> We have an app that uses:
> 
> ```
>   just_audio: ^0.9.34
>   just_audio_background: ^0.0.1-beta.10
> ```
> 
> _(and so also indirectly depends on their dependent audio related packages)._
> 
> We upgraded to Flutter `3.13.0-stable` on Wed Aug 16. And since then, **we started seeing crashes (full app crash)** on **iPhone 14 Pro (running iOS 16.6)**. More details on the patterns we have seen below, but essentially:
> 
> * we have only seen it fail on iPhone 14 Pros. We have tried two iPhone 14 Pros and it crashes consistently on both!
> * we also are testing on iPhone 12 Pro and iPhone 13 Pro, as well as Andriod (Samsung Galaxy A52) - no crashes so far.
> * the app seems to crash when running in background mode while audio playing
> * `Crashlytics`, or `background_audio`'s error handler are not seeing any of these crashes. We have this code to detect errors from `background_audio` (as well as comprehensive Crashlytics):
> 
> ```dart
>     _player.playbackEventStream.listen((event) {},
>         onError: (Object e, StackTrace stackTrace) => errorReportingService
>             .reportError('Error in playback stream', e, stackTrace));
> ```
> 
> (_The `errorReportingService` is configured to log it and also send to Crashlytics, and we know it works because we see other errors in Crashlytics, as expected_)
> 
> I downgraded Flutter back to `3.10.6-stable` today and have not seen any crashes so far.
> 
> To reiterate:
> 
> * we did not have crashes (on any device, including iPhone 14 Pros) before AUg 16th (on Flutter `3.10.6-stable`)
> * we have consistent crashes on iPhone 14 Pros with the `3.13.0-stable` release
> * after downgrading Flutter back to `3.10.6-stable` earlier today (a few hours ago), we have not seen crashes again. (But yes, that terrible lagging/jank is back :()
> 
> ### Patterns on failures on iPhone 14 Pro + iOS 16.6
> From @ilyabezdelev (who is seeing this in two iPhone 14 Pros), patterns when it crashes:
> 
> * iPhone locked, audio is playing in the background => the app crashes after a few minutes of playing. Sometimes it crashes a few seconds after locking the phone. Sometimes it may take 20-ish minutes before it crashes.
> * iPhone unlocked, switching to another app => the app crashes within a minute or so.
> * iPhone unlocked, app is open, auto-lock is set to 5 mins => app crashes immediately after the phone is auto-locked
> 
> It does not seem to crash when the app is in foreground.
> 
> ### Crash reports
> Surprisingly, Crashlytics does not show any of these crashes. I have even instrumented everything instructed in Crashlytics docs:
> 
> * [uncaught](https://firebase.google.com/docs/crashlytics/customize-crash-reports?hl=en&authuser=0&platform=flutter#report-uncaught-exceptions), including async ones; and [caught exceptions](https://firebase.google.com/docs/crashlytics/customize-crash-reports?hl=en&authuser=0&platform=flutter#report-caught-exceptions)
> * and even [Errors outside of Flutter Isolate](https://firebase.google.com/docs/crashlytics/customize-crash-reports?hl=en&authuser=0&platform=flutter#errors-outside-flutter)
> 
> We do see TestFlight crash reports, but they don't have any stack-trace or any error message.
> 
> ### Repro
> _PS: Our app is in TestFlight (and not launched yet), so I am unable to share code or more details. However, I will try and also create a minimal repro app if folks want it._

Same issues on iPhone 13 Pro Max.

Like this https://github.com/flutter/flutter/issues/106277

**Comment by luckysmg** [comment]
https://github.com/flutter/flutter/issues/132666#issuecomment-1685158321

**Comment by naivetoby** [comment]
"I am facing an iOS crash issue that I'd like to describe to the Flutter community. This problem can be consistently reproduced on any page, not just limited to CustomScrollView and Sliver widgets such as grids and lists.

Specifically, the CustomScrollView seems to trigger off-screen rendering, also known as the raster thread, more frequently during scrolling. The key point of reproduction lies in one of the pages of our app. When rapidly scrolling on this page while simultaneously switching between multiple other apps, the crash occurs after repeating this process several times.

It appears that the crash is somehow related to being present within the raster thread."

**Comment by stuartmorgan** [comment]
Please file a new issue for anything that is not the scroll jank in iOS 16.6 that this issue is tracking.

**Comment by arnab** [comment]
@stuartmorgan sounds good. There is already https://github.com/flutter/flutter/issues/132666 describing the issue.

**Comment by arnab** [comment]
Hi @stuartmorgan - I get that these above few comments are off-topic. However, I do think you (the Flutter team) should leave a warning of some sort in this post (such as this post - and I'll make it brief).

Like a lot of other developers, I landed on this story because I noticed significant lag in my app after I upgraded to iOS 16.6 (the original issue for which this issue was opened) - and saw that there was a fix for it in the latest `3.13.0-stable` release! Awesome, I upgraded. And then I noticed (only via user reports, since it doesn't fail in Simulator, or in my own devices) that **my app was crashing consistently when it was in the background** with this **Flutter 3.13.0-stable** release (the latest available and recommended release). IMO, this is an even more severe issue with this release.

**I do think by hiding the above posts (even though it is linked to #132666) the Flutter team is doing a dis-service to developers, who will land on this issue, see that there is a stable release that fixes the lag problem on iOS 16.6 and upgrade (and then may even release their apps, which will only crash on some user's devices; making their apps essentially unusable for those devices).**

Request you to add some sort of note here (or somewhere), since the latest "stable" release is not stable and should not be recommended by the Flutter team. Ultimately it's your framework - so you own the decision, but please consider the above. I think you should err on the side of being more transparent about such cases, rather than optimize for your rules of keeping each issue specific to a problem (which is generally a good idea)!

**Comment by jonahwilliams** [comment]
@arnab all you are doing by continually posting in this issue is making it difficult for us to triage and fix these bugs. If you do not stop posting off topic after being repeatly asked to cease I will escalate the matter as a CoC violation.

**Comment by arnab** [comment]
Wow! Didn't expect that as a response from the Flutter team!

> all you are doing by continually posting in this issue is making it difficult for us to triage and fix these bugs.

How so? It is simply a request to leave the comment up in this thread, so that other developers can read it - and decide for themselves if they should upgrade to `3.13.0-stable` or not. That is all! You don't need to respond or do anything else with my comment. You don't even need to leave my comment up. Like I said, just simply add a note here!

> If you do not stop posting off topic after being repeatly asked to cease I will escalate the matter as a CoC violation.

I am sincerely sorry to hear this! My aim is not distract the Flutter team (you don't need to respond to this at all), but simply to help other Flutter developers/users.

---

If you are going to ban me, please also educate me about how my comments are violating the Code of Conduct? 

On the contrary, from https://dart.dev/community/code-of-conduct:

> We expect Dart community members to act professionally and respectfully, and we expect our social spaces to be safe and dignified environments.

Your comment (and threat) does not make me feel like this is a safe space for me to continue posting.

**Comment by itsjustkevin** [comment]
For those affected by the iOS 16.6 performance regression and are not ready to upgrade to Flutter 3.13 yet, good news, we have a fix!

We were able to backport a fix to the Flutter 3.10 series.  

For guidance on how to update to Flutter [3.10.7](https://github.com/flutter/flutter/releases/tag/3.10.7) please take a look at our [wiki entry](https://github.com/flutter/flutter/wiki/Upgrading-from-3.10.6-to-3.10.7).

**Comment by zanderso** [comment]
Thanks @itsjustkevin. I believe there's no further action to take on this issue, so I will close it.
